txrelayer/txrelayer.go

```diff
@@ -72,6 +72,18 @@ func NewTxRelayer(opts ...TxRelayerOption) (TxRelayer, error) {
 		opt(t)
 	}
 
+	// Calculate receiptsPollFreq based on receiptsTimeout
+
+	if t.receiptsTimeout >= time.Minute {
+		t.receiptsPollFreq = 2 * time.Second
+	}
+
+	if t.receiptsPollFreq >= t.receiptsTimeout || t.receiptsTimeout < time.Second {
```

Comment:
Not good enough. If your receiptsTimeout is lets say 50s, this will only ping the rpc node twice.
You need a condtion if t.receiptsTimeout >= time.Second, if that is the case, set it to 1s.

Reply:
I disagree with that statement because I have already set `receiptsPollFreq` to one second as the default value (line 68). Therefore, I don't see the need for that condition.

Reply:
Ah, didn't see that change. Nice.

Reply:
Again, we are making txRelayer smarter than it should be IMHO.

I'd just return an error in case `t.receiptsPollFreq >= t.receiptsTimeout` and that's all. Let the caller take care which values it provides, but they just need to be valid.